core/sql/optimizer/ItemExpr.cpp

```diff
@@ -14906,3 +14906,80 @@ NABoolean LOBoper::isCovered
   return FALSE;
 }
 
+// Compute the exprssion at compile time. Assume all operands are constants.
+// Return NULL if the computation fails and CmpCommon::diags() may be side-affected.
+ConstValue* ItemExpr::compute(CollHeap* heap)
+{
+  ValueIdList exprs;
+  exprs.insert(getValueId());
+
+  const NAType& dataType = getValueId().getType();
+
+  Lng32 decodedValueLen = dataType.getNominalSize() + dataType.getSQLnullHdrSize();
+
+  char staticDecodeBuf[200];
+  Lng32 staticDecodeBufLen = 200;
+
+  char* decodeBuf = staticDecodeBuf;
+  Lng32 decodeBufLen = staticDecodeBufLen;
+
+  // For character types, multiplying by 8 to deal with conversions between
```

Comment:
Good point. I'll fix it.

On Mon, Oct 19, 2015 at 1:02 PM, DaveBirdsall <notifications@github.com>
wrote:

> In core/sql/optimizer/ItemExpr.cpp
> <https://github.com/apache/incubator-trafodion/pull/132#discussion_r42404547>
> :
>
> > +ConstValue* ItemExpr::compute(CollHeap* heap)
> > +{
> > +  ValueIdList exprs;
> > +  exprs.insert(getValueId());
> > +
> > +  const NAType& dataType = getValueId().getType();
> > +
> > +  Lng32 decodedValueLen = dataType.getNominalSize() + dataType.getSQLnullHdrSize();
> > +
> > +  char staticDecodeBuf[200];
> > +  Lng32 staticDecodeBufLen = 200;
> > +
> > +  char* decodeBuf = staticDecodeBuf;
> > +  Lng32 decodeBufLen = staticDecodeBufLen;
> > +
> > +  // For character types, multiplying by 8 to deal with conversions between
>
> UTF-8 characters are a max of 6 bytes. Do we know of any that are longer?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/132/files#r42404547>.
>



-- 
Regards, --Qifan